.github/workflows/main.yml

```diff
@@ -0,0 +1,42 @@
+name: CI
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ master, develop ]
```

Comment:
If this is purely to generate docs, I don't think it should run on pull requests, right? That way a "bad pr's docs" won't overwrite the published version. However, to test this pull request in particular, it could be useful.

Reply:
I agree. I wasnt thinking. Arguably it really should just be pushes to master or develop (even though master is protected) and merges to master.

Reply:
Just learned merges implictly cause a push even so just push to master or develop should be fine. Additionally the update to "docs" dont occur on the branch pushed. They occur TO the gh-pages branch based off the source in the new merged version of master or develop.